File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);
        int max = people.length - 1;
        for(int i = 0; i <= max; i++){
            if(people[i] + people[max] > limit){
                i--;
            }
            max--;
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var answer = 0;
    people.sort((a,b) => a - b);
    let max = people.length - 1;
    for(let i = 0; i <= max; i++){
        if(people[i] + people[max] > limit) i--;
        max--;
        answer++;
    }
    return answer;
}